code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -185,10 +188,11 @@ All foods are distributed among various categories. Use common sense.
 			var/obj/item/reagent_containers/food/snacks/customizable/C = new custom_food_type(get_turf(src))
 			C.initialize_custom_food(src, S, user)
 			return 0
-	var/sharp = W.get_sharpness()
-	if(sharp)
-		if(slice(sharp, W, user))
-			return 1
+	if(user.a_intent != INTENT_DISARM)
+		var/sharp = W.get_sharpness()
+		if(sharp)
+			if(slice(sharp, W, user))
+				return 1
```

Comment:
because you are already in the scope of if(sharp) when you do those conditions, if its false it just returns ``.`` which means we can just return the result of this as we don't actually use the value of ``.``

```suggestion
		return sharp && slice(sharp, W, user)
```